File: lib/antlr-2.7.5/examples/csharp/heteroAST/INTNode.cs
using System;
using System.IO;
using antlr;
using antlr.collections;

/** A simple node to represent an INT */
public class INTNode : CalcAST {
	int v=0;

	public INTNode() {
	}

	public INTNode(Token tok) {
		v = Convert.ToInt32(tok.getText());
	}

	public override void initialize(int t, String txt) {
		if (t == CalcParserTokenTypes.INT)
			v = Convert.ToInt32(txt);
	}
	public override void initialize(AST t) {
		if (t.Type == CalcParserTokenTypes.INT)
			v = Convert.ToInt32(t.getText());
	}
	public override void initialize(IToken tok) {
		if (tok.Type == CalcParserTokenTypes.INT)
			v = Convert.ToInt32(tok.getText());
	}
 
 	/** Compute value of subtree; this is heterogeneous part :) */
	public override int Value() {
		return v;
	}

	public override string ToString() {
		return " "+v;
	}

	public override void xmlSerializeNode(TextWriter outWriter) {
		outWriter.Write("<int>"+v+"</int>");
	}
}

File: lib/antlr-2.7.5/examples/java/heteroAST/INTNode.java
import antlr.BaseAST;
import antlr.Token;
import antlr.collections.AST;
import java.io.*;

/** A simple node to represent an INT */
public class INTNode extends CalcAST {
	int v=0;

	public INTNode() {
	}

	public INTNode(Token tok) {
		v = Integer.parseInt(tok.getText());
	}

	/** Compute value of subtree; this is heterogeneous part :) */
	public int value() {
		return v;
	}

	public String toString() {
		return " "+v;
	}

	public void xmlSerializeNode(Writer out) throws IOException {
		out.write("<int>"+v+"</int>");
	}
}
